src/correlator_bosonic_recursive.jl

```diff
@@ -167,40 +179,86 @@ function add_operator_fermi(
       L_prev = (op_ind > 1 ? delta(dag(ln[op_ind - 1])', ln[op_ind - 1]) : 1.0) #initialize left environment
     end
 
-    L = L_prev#copy(L_prev) #copy cached environment after applying previous operator (this can be simplified)
+    L = L_prev  #copy(L_prev) #copy cached environment after applying previous operator (this can be simplified)
     op_psi = psi[op_ind]
 
     if jw % 2 != 0
       op_psi = apply(op("F", s[op_ind]), op_psi) #apply jordan wigner string if needed
     end
-    op_psi = apply(op(ops[counter], s[op_ind]), op_psi)  #apply operator in the spot #counter
-
-    jw_next = 0
-    if ops[counter] == "Cdagup" ||
-      ops[counter] == "Cdag" ||
-      ops[counter] == "Cup" ||
-      ops[counter] == "C"
-      jw_next = jw + 1
-      op_psi = apply(op("F", s[op_ind]), op_psi) #track if a fermionic operator was applied
-    elseif ops[counter] == "Cdagdn" || ops[counter] == "Cdn"
-      jw_next = jw + 1
-      op_psi = apply(op("F", s[op_ind]), op_psi) #for spin down operator we need a j-w term on-sitw
+
+    for i in 0:repeat
+      op_psi = apply(op(ops[perm_ind[counter + repeat - i]], s[op_ind]), op_psi)
+    end
+
+    jw_next = jw
+    for i in 0:repeat
+      if ops[perm_ind[counter + repeat - i]] == "Cdagup" ||
+        ops[perm_ind[counter + repeat - i]] == "Cdag" ||
+        ops[perm_ind[counter + repeat - i]] == "Cup" ||
+        ops[perm_ind[counter + repeat - i]] == "C"
```

Comment:
I think we should use `ITensorMPS.has_fermion_string` as discussed elsewhere in this PR instead of these hardcoded checks.